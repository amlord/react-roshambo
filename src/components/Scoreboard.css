.Scoreboard {
    vertical-align: top;
    font-size: 1rem;
    display: inline-block;
    width: 31.578947368421053%;
    margin-right: 2.631578947368421%;
}

.Scoreboard__scores {
    font-size: 0;
}

.Scoreboard__score {
    font-size: 1rem;
    display: inline-block;
    width: 27.777777777777778%;
    margin-right: 8.333333333333333%;
}

.Scoreboard__score:last-child {
    margin-right: 0;
}