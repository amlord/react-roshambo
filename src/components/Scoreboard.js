import React, { Component } from 'react';
import logo from '../logo.svg';
import './Scoreboard.css';

class Scoreboard extends Component {
  render() {
    return (
      <div className="Scoreboard">Scoreboard Component</div>
    );
  }
}

export default Scoreboard;
