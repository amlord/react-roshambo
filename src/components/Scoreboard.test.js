import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import Scoreboard from './Scoreboard';

describe('Scoreboard', () => {
  const mockReset = jest.fn();
  const props = { score: { win: 0, draw: 0, loss: 0 }, resetGame: mockReset };
  const scoreboard = shallow(<Scoreboard {...props} />);

  it('renders correctly', () => {
    expect(scoreboard).toMatchSnapshot();
  });

  describe('when clicking the `reset` button', () => {
    beforeEach( () => {
      scoreboard.find('.Scoreboard__reset').simulate('click');
    });

    it('calls the `resetGame` callback', () => {
      expect(mockReset).toHaveBeenCalled();
    });
  });

  describe('when the score changes', () => {
    it('`win` score label updates', () => {
      expect( scoreboard.find('.Scoreboard__scoreCount--win').text() ).toEqual("0");

      scoreboard.setProps( { score: { win: 1, draw: 0, loss: 0 } } );

      expect( scoreboard.find('.Scoreboard__scoreCount--win').text() ).toEqual("1");
    });

    it('`draw` score label updates', () => {
      expect( scoreboard.find('.Scoreboard__scoreCount--draw').text() ).toEqual("0");

      scoreboard.setProps( { score: { win: 0, draw: 2, loss: 0 } } );

      expect( scoreboard.find('.Scoreboard__scoreCount--draw').text() ).toEqual("2");
    });

    it('`loss` score label updates', () => {
      expect( scoreboard.find('.Scoreboard__scoreCount--loss').text() ).toEqual("0");

      scoreboard.setProps( { score: { win: 0, draw: 0, loss: 3 } } );

      expect( scoreboard.find('.Scoreboard__scoreCount--loss').text() ).toEqual("3");
    });
  });
});