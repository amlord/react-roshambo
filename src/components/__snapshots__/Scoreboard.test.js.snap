// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Scoreboard
    resetGame={[Function]}
    score={
        Object {
            "draw": 0,
            "loss": 0,
            "win": 0,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
                    Scoreboard
          </h1>
</header>,
        <div
          className="Scoreboard__scores"
>
          <div
                    className="Scoreboard__score Scoreboard__score--win"
          >
                    <h2>
                              Win
                    </h2>
                    <div
                              className="Scoreboard__scoreCount Scoreboard__scoreCount--win"
                    />
          </div>
          <div
                    className="Scoreboard__score Scoreboard__score--draw"
          >
                    <h2>
                              Draw
                    </h2>
                    <div
                              className="Scoreboard__scoreCount Scoreboard__scoreCount--draw"
                    />
          </div>
          <div
                    className="Scoreboard__score Scoreboard__score--loss"
          >
                    <h2>
                              Loss
                    </h2>
                    <div
                              className="Scoreboard__scoreCount Scoreboard__scoreCount--loss"
                    />
          </div>
</div>,
        <button
          className="btn--reset"
          onClick={[Function]}
>
          Reset Game
</button>,
        <ScoreHistory
          history={undefined}
/>,
      ],
      "className": "Scoreboard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Scoreboard
</h1>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Scoreboard",
          },
          "ref": null,
          "rendered": "Scoreboard",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Scoreboard__score Scoreboard__score--win"
>
              <h2>
                            Win
              </h2>
              <div
                            className="Scoreboard__scoreCount Scoreboard__scoreCount--win"
              />
</div>,
            <div
              className="Scoreboard__score Scoreboard__score--draw"
>
              <h2>
                            Draw
              </h2>
              <div
                            className="Scoreboard__scoreCount Scoreboard__scoreCount--draw"
              />
</div>,
            <div
              className="Scoreboard__score Scoreboard__score--loss"
>
              <h2>
                            Loss
              </h2>
              <div
                            className="Scoreboard__scoreCount Scoreboard__scoreCount--loss"
              />
</div>,
          ],
          "className": "Scoreboard__scores",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Win
</h2>,
                <div
                  className="Scoreboard__scoreCount Scoreboard__scoreCount--win"
/>,
              ],
              "className": "Scoreboard__score Scoreboard__score--win",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Win",
                },
                "ref": null,
                "rendered": "Win",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "Scoreboard__scoreCount Scoreboard__scoreCount--win",
                },
                "ref": null,
                "rendered": 0,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Draw
</h2>,
                <div
                  className="Scoreboard__scoreCount Scoreboard__scoreCount--draw"
/>,
              ],
              "className": "Scoreboard__score Scoreboard__score--draw",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Draw",
                },
                "ref": null,
                "rendered": "Draw",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "Scoreboard__scoreCount Scoreboard__scoreCount--draw",
                },
                "ref": null,
                "rendered": 0,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h2>
                  Loss
</h2>,
                <div
                  className="Scoreboard__scoreCount Scoreboard__scoreCount--loss"
/>,
              ],
              "className": "Scoreboard__score Scoreboard__score--loss",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Loss",
                },
                "ref": null,
                "rendered": "Loss",
                "type": "h2",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                  "className": "Scoreboard__scoreCount Scoreboard__scoreCount--loss",
                },
                "ref": null,
                "rendered": 0,
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Reset Game",
          "className": "btn--reset",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Reset Game",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "history": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
                        Scoreboard
            </h1>
</header>,
          <div
            className="Scoreboard__scores"
>
            <div
                        className="Scoreboard__score Scoreboard__score--win"
            >
                        <h2>
                                    Win
                        </h2>
                        <div
                                    className="Scoreboard__scoreCount Scoreboard__scoreCount--win"
                        />
            </div>
            <div
                        className="Scoreboard__score Scoreboard__score--draw"
            >
                        <h2>
                                    Draw
                        </h2>
                        <div
                                    className="Scoreboard__scoreCount Scoreboard__scoreCount--draw"
                        />
            </div>
            <div
                        className="Scoreboard__score Scoreboard__score--loss"
            >
                        <h2>
                                    Loss
                        </h2>
                        <div
                                    className="Scoreboard__scoreCount Scoreboard__scoreCount--loss"
                        />
            </div>
</div>,
          <button
            className="btn--reset"
            onClick={[Function]}
>
            Reset Game
</button>,
          <ScoreHistory
            history={undefined}
/>,
        ],
        "className": "Scoreboard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Scoreboard
</h1>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Scoreboard",
            },
            "ref": null,
            "rendered": "Scoreboard",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Scoreboard__score Scoreboard__score--win"
>
                <h2>
                                Win
                </h2>
                <div
                                className="Scoreboard__scoreCount Scoreboard__scoreCount--win"
                />
</div>,
              <div
                className="Scoreboard__score Scoreboard__score--draw"
>
                <h2>
                                Draw
                </h2>
                <div
                                className="Scoreboard__scoreCount Scoreboard__scoreCount--draw"
                />
</div>,
              <div
                className="Scoreboard__score Scoreboard__score--loss"
>
                <h2>
                                Loss
                </h2>
                <div
                                className="Scoreboard__scoreCount Scoreboard__scoreCount--loss"
                />
</div>,
            ],
            "className": "Scoreboard__scores",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Win
</h2>,
                  <div
                    className="Scoreboard__scoreCount Scoreboard__scoreCount--win"
/>,
                ],
                "className": "Scoreboard__score Scoreboard__score--win",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Win",
                  },
                  "ref": null,
                  "rendered": "Win",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "Scoreboard__scoreCount Scoreboard__scoreCount--win",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Draw
</h2>,
                  <div
                    className="Scoreboard__scoreCount Scoreboard__scoreCount--draw"
/>,
                ],
                "className": "Scoreboard__score Scoreboard__score--draw",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Draw",
                  },
                  "ref": null,
                  "rendered": "Draw",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "Scoreboard__scoreCount Scoreboard__scoreCount--draw",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "div",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h2>
                    Loss
</h2>,
                  <div
                    className="Scoreboard__scoreCount Scoreboard__scoreCount--loss"
/>,
                ],
                "className": "Scoreboard__score Scoreboard__score--loss",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Loss",
                  },
                  "ref": null,
                  "rendered": "Loss",
                  "type": "h2",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 0,
                    "className": "Scoreboard__scoreCount Scoreboard__scoreCount--loss",
                  },
                  "ref": null,
                  "rendered": 0,
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Reset Game",
            "className": "btn--reset",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Reset Game",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "history": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
